### ISSUE TITLE
5V drivers logic

### ISSUE THREAD

**Issue by rafaljot** [bobc/bobc_hardware#33]
I have S109 drivers based on TB67S109AFTG. They need vcc=5V.
May I set VLOG JP101 to 3-2, (FD_5V) with Due ?

**Comment by bobc** [comment]
> May I set VLOG JP101 to 3-2, (FD_5V) with Due ?

No, do not do that, it may damage your Due. Only use JP101 pins 2-3 if you have a 5V baseboard like Arduino Mega AND the Mega is an old version that does not supply 5V on IOREF.

> They need vcc=5V.

Really? According to the TB67S109A datasheet https://toshiba.semicon-storage.com/info/docget.jsp?did=14642&prodName=TB67S109AFTG, VCC is an output from the TB67S109A's internal regulator.

I don't know what modules you have, but according to https://www.aliexpress.com/store/product/4pcs-Lot-3D-Printer-Stepstick-S109-Stepper-Motor-Driver-Compatible-with-57-Stepper-Motor/1757194_32809825741.html the modules will work with 3.3V logic.

If you have those modules, they should work with Due at 3.3V.

**Comment by rafaljot** [comment]
Thank you!
Yes I have this one. It works with 5V,  with Mega. in my Velleman Vertex K8400. With FD it looks like only EN works and ignores STEP/DIR.
BTW: not bed solutions for 2.5amps motors, for extruder.

**Comment by bobc** [comment]
The usual problem with Due is the pulse width is much shorter than AVR, because of the faster CPU. Allegro A4988 etc drivers can cope with short (1us) pulse width, but other drivers need a longer pulse, According to data sheet for 'S109 the minimum pulse is 10us. With whatever firmware you are using, try increasing the pulse width on the STEP signal.
